src/main/scala/com/monsanto/arch/cloudformation/model/resource/AutoScaling.scala

```diff
@@ -10,12 +10,13 @@ import spray.json._
 
 case class `AWS::AutoScaling::AutoScalingGroup`(
```

Comment:
Looks like you shouldn't be specifying both AvailabilityZones and VPCZoneIdentifier. For maximum type safety we should make both optional and make the whole constructor private. Then we can add 2 new methods that instantiate a ASG with the different methods as well as a default for backwards compatibility in the companion. See AWS::EC2::EIP as an example.